File: test/validation.m
data = rand(50, 5);
% data = load('/home/adriano/workspace/subspace-learning/data/mammals1k.data');
X = data';
A = X;
l = 0.1;

addpath('/home/adriano/workspace/subspace-learning/lrr');
[Z, E] = solve_lrr(X, A, l, 1, 1, true);

save('/tmp/data', 'data', '-ascii');

system('python /home/adriano/workspace/lrr/test/validation.py');

ZZ = load('/tmp/Z');
EE = load('/tmp/E');

dif_Z = max(max(abs(Z - ZZ)));
dif_E = max(max(abs(E - EE)));
disp(['errors: ', num2str(dif_Z), ' ', num2str(dif_E)])

File: test/validation.py
import numpy as np
import sys
sys.path.insert(0, '/home/adriano/workspace/lrr')

from solve_lrr import solve_lrr

data = np.loadtxt('/tmp/data')
X = data.T
A = X
l = 0.1

Z, E = solve_lrr(X, A, l, reg=1, alm_type=1, display=True)

np.savetxt('/tmp/Z', Z)
np.savetxt('/tmp/E', E)
